firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -627,6 +627,21 @@ extension BrowserViewController: WKNavigationDelegate {
             } else {
                 webView.customUserAgent = UserAgent.getUserAgent(domain: url.baseDomain ?? "")
             }
+
+            if navigationAction.navigationType == .linkActivated {
+                if tab.isPrivate {
+                    decisionHandler(.cancel)
+                    webView.load(navigationAction.request)
+                    return
+                } else if UIApplication.shared.canOpenURL(url) {
+                    UIApplication.shared.open(url, options: [.universalLinksOnly: true]) { isAppInstalled in
+                        if isAppInstalled {
+                           // TODO: https://mozilla-hub.atlassian.net/browse/FXIOS-7524
+                        }
+                    }
+                }
+            }
+
             decisionHandler(.allow)
             return
         }
```

Comment:
Yes, we definitely can, but this would mean repeating this code at the end of the `else if` block, in order to maintain the same logic:

`decisionHandler(.allow)`
`return`

Reply:
hmm, probably fine to keep it then :)